tests/tests3.cpp

```diff
@@ -385,3 +385,62 @@ TEST_CASE("Escape chars") {
     storage.update(selena);
     storage.remove<Employee>(10);
 }
+
+#ifdef SQLITE_ORM_WITH_CTE
+TEST_CASE("With select") {
+    using Catch::Matchers::Equals;
+
+    using cnt = decltype(1_ctealias);
+    auto storage = make_storage("");
+    SECTION("with ordinary") {
+        auto rows = storage.with(cte<cnt>().as(select(1)), select(column<cnt>(1_colalias)));
+        REQUIRE_THAT(rows, Equals(std::vector<int>{1}));
```

Comment:
IMO it is idiomatically better to move `REQUIRE_THAT` to the end of the test case and to have `std::vector<int> rows` and `std::vector<int> expected` in the beginning if the test case, then assign `rows` and `expected` in every section. Such a structure shows that requirements in all sections are the same

Reply:
IMO it is more idiomatic if each section is checked for its own requirement. Otherwise - and I've come across this problem many times in sqlite_orm - you won't find the wrong section if the assertion is made at test case level.

Also, looking at the short documentation on [test cases and sections](https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md), I would conclude that it is good to assert common things across all sections at the test case level, but each section asserts its own things.

Reply:
IMO it is more idiomatic if each section is checked for its own requirement. Otherwise - and I've come across this problem many times in sqlite_orm - you won't find the wrong section if the assertion is made at test case level.

Also, looking at the short documentation on [test cases and sections](https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md), I would conclude that it is good to assert common things before all sections at the test case level, but each section asserts its own things.

Reply:
> I've come across this problem many times in sqlite_orm - you won't find the wrong section if the assertion is made at test case level.

Yes it is definitely a problem with sections but one can fix this by adding special `if`s or putting a breakpoint right inside the section, then back to assertion. This is how I do it.

> Also, looking at the short documentation on [test cases and sections](https://github.com/catchorg/Catch2/blob/devel/docs/test-cases-and-sections.md), I would conclude that it is good to assert common things before all sections at the test case level, but each section asserts its own things.

I disagree here cause those docs show toy examples like OOP's `Shape`, `Rectangle` and `Triangle` classes which are never used that way. So I see nothing bad in putting some `REQUIRE` after all the sections cause this is one of genuine way of using sections

Reply:
Yes, I also do search for the exact failing section just like you do. And I don't say it's a wrong way of using section.
But imagine you have dozens of sections, it's cumbersome. I want to see the failing code line right in the test output. Imagine a compiler wouldn't tell you the line of the wrong code. So I disagree. What shall we do?

Reply:
let's leave it as you want it to be cause it is your PR and the topic of this conversations is more personal preferences not something critical